### ISSUE TITLE
deletion of backup files results in a disaster

### ISSUE THREAD

**Issue by honghaoz** [lra/mackup#1127]
I thought backup files are just backups at first.

However, it's not. All of backup files are symbolic linked and no original files are kept.

Now I accidentally removed backups and nothing left...

This is a disaster for me...

I think you should mention backup files are moved and linked. This is dangerous.

**Comment by fr0der1c** [comment]
I agree with you. In fact, current behaviour of mackup can cause other problem like #1149

**Comment by slaughtr** [comment]
The README is very unclear on this. It definitely does mention `ln -s` commands, and has the word `symlink` one time...but I assumed the symlinks went the other way to build some kind of app index for mackup. This totally screwed my day since most of my dev tool configs are on a hard drive elsewhere...and all I wanted to do was copy settings from one laptop to another.

It makes sense that it works the way it does, the README just needs a big, bold mention of 'THIS WILL REMOVE ALL YOUR FILES AND CREATE SYMLINKS IN THEIR PLACE'

**Comment by fr0der1c** [comment]
I agree with @slaughtr . In fact, I would say  the current `mackup backup` should be renamed to `mackup sync`, and we should have a real `mackup backup`. Syncing all preferences between multiple devices is not we need at all times.

**Comment by fullmetalsheep** [comment]
This just happened to me, all preferences of 3-4 apples vanished, haha.

**Comment by fr0der1c** [comment]
I think we should makes some changes and pull request. It's truly not what it expected to be.

**Comment by fullmetalsheep** [comment]
The symlink is the main issue. There should be a warning not to delete the backup files once deleted. Or mackup should be re-created from scratch using rsync and rclone as the backup methods.

**Comment by fr0der1c** [comment]
I think the original `mackup backup` command should be named `mackup sync` instead. And we should have a REAL backup feature called `backup`.

**Comment by ikari-pl** [comment]
It gets even more fun when my synced folder unsyncs (I use odrive) and random apps lose settings. To make it more spicy, "mackup uninstall" fails with a `UnicodeEncodeError: 'ascii' codec can't encode character u'\xdc' in position 28: ordinal not in range(128)` for me, so if I want to un-symlink, it looks like a long manual process.

**Comment by fullmetalsheep** [comment]
@ikari-pl 

> In /usr/local/lib/python2.7/dist-packages/backports/configparser/__init__.py, I changed this line optname = self.optionxform(optname.rstrip()) to optname = self.optionxform(optname.encode('utf-8').rstrip()) and the problem is solved.

Above should fix it. Even I had the same issue. Furthermore, I have setup rsync and it works nicely as intended without all the symlink messes. Please watch my repo if you're still interested. I'll update the fork asap.

**Comment by thsowers** [comment]
+1 to the number of frustrated users who have encountered this.

From the README: 

> `mackup backup` Backup your application settings.

> Back ups your application settings in a safe directory (e.g. Dropbox)

But this doesn't _technically_ happen at all (sync != backup) 

As other users have pointed out, this repo needs a large mention that this will remove _all_ your files and create symlinks in there place. Very misleading and disappointing.

**Comment by esetnik** [comment]
The default should never replace existing files with symlinks. This is almost never what the user wants to do, especially in a backup scenario. You can never be sure that a remote service or device will be accessible. Most of us just want to manually trigger a backup and restore.

**Comment by paxperscientiam** [comment]
The problem with doing it the "other way around" is that cloud services handle symlinks differently.

The service might just backup the symlink as is, but what's the point? If your machine dies, then you're left with useless symlinks to nowhere.

The service might follow the symlink, but then you wind up with a static copy in your drive.

While I agree that the language needs to be cleared up, I for one prefer syncing over backup. For one, backing up is either going to require user to manually do backup periodically, or configure cron job, or mackup would have to include a daemon.

**Comment by lra** [comment]
I'm totally open for improvement over the language until we have a technical solution: PR welcomed!

**Comment by jamiew** [comment]
+1 on code being most useful way to move forward with this. I would happily help test a `sync` process that copied rather than symlinked files

**Comment by rinkjames** [comment]
This just happened to me... I was trying to find a way to backup important config files and settings and in the process of trying out this option Mackup effectively _moved_ all my files to my specified test directory. I checked the directory to see what files I understood it had only _copied_. "Great, so this is what it'll do", I thougt. I proceeded to remove the test backup so that I could try some other options. Suddenly nothing works. I just deleted trashed the very files I was trying to backup. Fuck.

**Comment by joshmedeski** [comment]
I highly recommend you put your dotfiles in version control. That way if mackup acts up and deletes files you can restore them with version control.

As for this conversation, I'd say some of this could be cleared up with an FAQ section in the README, or a more in-depth explanation of what the commands do.

I'm thinking about writing an article talking in-depth on how mackup's commands work and how I use it. Leave a 👍  if you'd like to see that article.

**Comment by joshmedeski** [comment]
@vfonic has an interesting suggestion for making the functionality more clear, see https://github.com/lra/mackup/issues/1693#issuecomment-775420851
